prov/sockets/src/sock_poll.c

```diff
@@ -55,6 +58,24 @@ static int sock_poll_add(struct fid_poll *pollset, struct fid *event_fid,
 	list_item->fid = event_fid;
 	dlist_init(&list_item->entry);
 	dlist_insert_after(&list_item->entry, &poll->fid_list);
+
+	switch (list_item->fid->fclass) {
+	case FI_CLASS_CQ:
+		cq = container_of(list_item->fid, struct sock_cq, cq_fid);
+		atomic_inc(&cq->ref);
+		break;
+	case FI_CLASS_CNTR:
+		cntr = container_of(list_item->fid, struct sock_cntr, cntr_fid);
+		atomic_inc(&cntr->ref);
+		break;
+	case FI_CLASS_EQ:
+		eq = container_of(list_item->fid, struct sock_eq, eq);
```

Comment:
Does the code take reference counts in both directions?  I.e. the CQ on the poll set and the poll set on the CQ.